app/src/main/java/com/hedvig/app/feature/offer/ui/changestartdate/ChangeDateBottomSheet.kt

```diff
@@ -37,94 +52,96 @@ class ChangeDateBottomSheet : BottomSheetDialogFragment() {
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-
-        binding.apply {
-            datePickButton.setHapticClickListener {
-                showDatePickerDialog()
+        lifecycleScope.launch {
+            changeDateBottomSheetViewModel.viewState.collect { viewState ->
```

Comment:
Collection of `StateFlow` requires some special care, I believe, in order to function properly with android lifecycles. Check out https://developer.android.com/kotlin/flow/stateflow-and-sharedflow#stateflow (in particular, `repeatOnLifecycle`) 🙂

Reply:
Good catch, created an extension function for this